src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -226,22 +230,26 @@ public DRSObject lookupObjectByDrsIdPassport(
       String drsObjectId, DRSPassportRequestModel drsPassportRequestModel) {
     try (DrsRequestResource r = new DrsRequestResource()) {
       DrsId resolvedDrsObjectId = resolveDrsObjectId(drsObjectId);
-      List<SnapshotCacheResult> cachedSnapshots =
+      List<Future<SnapshotCacheResult>> futures =
           lookupSnapshotsForDRSObject(resolvedDrsObjectId).stream()
-              .parallel()
               .map(
-                  s -> {
-                    try {
-                      // Only look at snapshots that the user has access to
-                      verifyPassportAuth(s.id, drsPassportRequestModel);
-                      return s;
-                    } catch (IamForbiddenException e) {
-                      return null;
-                    }
-                  })
-              .filter(Objects::nonNull)
+                  s ->
+                      executor.submit(
+                          () -> {
+                            try {
+                              // Only look at snapshots that the user has access to
+                              verifyPassportAuth(s.id, drsPassportRequestModel);
+                              return s;
+                            } catch (UnauthorizedException e) {
+                              return null;
+                            }
+                          }))
               .toList();
-
+      List<SnapshotCacheResult> cachedSnapshots =
+          FutureUtils.waitFor(futures).stream().filter(Objects::nonNull).toList();
```

Comment:
I've seen this pattern in a few places, maybe it should be built into `FutureUtils`? Possibly `waitForNonNull()` or make the `filter` method a parameter to `waitFor()`?

Reply:
That's a great idea.

Reply:
I looked throughout the repo, and I couldn't find anywhere that we'd actually want to keep null values.

So I'd like to change `FutureUtils.waitFor` to perform the null filtering, which simplifies nearly every call to it and ensures that subsequent calls will also get the same protection. I can update the docstring to reflect this and add a `@NonNull` annotation on the returned list element type as a courtesy.